auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -270,15 +270,15 @@ public void simple() {
         + "      jsonWriter.name(\"o\");\n"
         + "      oAdapter.write(jsonWriter, object.o());\n"
         + "      jsonWriter.name(\"p\");\n"
```

Comment:
Empty/absent properties are serialized as null, or the default value for wrapped primitives.
User-created objects will be empty/absent in the following cases:
1. `Optional.empty()` is passed to the static factory method (when not using builders).
2. When using builders, the setter is not called before building and default setters are not enabled on the type adapter.
3. When using builders, the setter is not called before building and default setters are enabled but the default setter is never invoked.